meeting-minutes/2023-06-13.md

```diff
@@ -0,0 +1,161 @@
+
+cwebber: I think in some ways, bridges *are* an implementation.
+
+tsyesika: I think bridges in some ways are in implementation, that's true, but I don't think that's what we should aim for.  And if we have things that are currently ossified, well bridges definitely serve a great purpose.  But to me what motivates me is that I've seen how something looks like Cap'N Proto, whether it speaks Syrup or Spritely's things but the premise of CapTP, I think that's great, I want lots of implementations.  The thing we're doing here isn't necessarily building a specification for us, that's great, but I hope we're defining a specification for *all of the next implementations of CapTP that come after us*.  And they need to speak the same concrete syntax.  It doesn't need to be the one Spritely implements
+
+cwebber: And there's an exponential cost to implementing bridges if we only do bridges, rather than try to do a shared protocol
+
+zenhack: Quadratic but yes
+
+markm: I agree, we all want there to be a shared protocol and we've been doing work on something concrete.  All of that's great.  But I didn't hear any controversy about the ordering of priorities, that if we don't have a coherent story about semantic round tripping, then we're in a much worse situation than needing to do bridging.  And I want to interject one more note which is that the way in which ordinary programmers interact, the world we're anticipating where captp is somehow broadly adopted, they don't write captp, they use objects that interact with objects, and the entire mechanism of captp is that at every endpoint there is a bridge or converter between language objects and the protocol.  So it's already the case that in order to have a multi-language protcol there are converters at every endpoint.  And so if there's more bridging going on, that's not necessarily that much more of an impediment.
+
+tsyesika: I think you're right Mark, and I certainly don't diasagree with your assertion we should standardize on abstract syntax, that's important, I think bridges play a role, but when producing a test suite, I would prefer it if it has a concrete syntax.  I think it makes sense for new implementations that haven't been written to speak the language.
+
+zenhack: One thing, we were talking about the notion of 3rd party handoffs and having different lower netlayers pushed down... I'd like to ask people to review.. Cap'N Proto, it's not battle tested to do this, but I'd like to specify a way to do 3rd party handoffs that's agnostic to any netlayer details.  I'd like to ask folks to read that, and that's something floated a few times.  It would probably be useful as a starting point of what it might look like.  It feels like that would be useful
+
+https://github.com/ocapn/ocapn/issues/53#issuecomment-1572439783
+https://github.com/capnproto/capnproto/blob/455824528cd01268d4f9491a0df961c9a3678852/c%2B%2B/src/capnp/rpc.capnp#L1411-L1512
```

Comment:
Both Agoric and Spritely agreed to look at rpc.capnp#L1411-L1512 . I'm struggling to come up with a concrete suggestion for how to say that in the notes.

It's partly water under the bridge: @zenhack already turned the "action items" into assignments on #53, which I think is a fine way to keep the ball moving.